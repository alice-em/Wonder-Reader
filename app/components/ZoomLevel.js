import PropTypes from 'prop-types';
import React from 'react';

const styles = {
  fontFamily: 'Carter One',
  fontSize: '20px',
  width: '45px',
  cursor: 'default',
};

const ZoomLevel = ({ value }) => (
  <div className="zoomLevel" style={styles}>
    {value}
  </div>
);

ZoomLevel.propTypes = {
  value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
};

export default ZoomLevel;
