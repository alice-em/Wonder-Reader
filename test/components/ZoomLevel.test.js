import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import React from 'react';

import ZoomLevel from '../../app/components/ZoomLevel';

Enzyme.configure({ adapter: new Adapter() });

describe('ZoomLevel', () => {
  it('should render', () => {
    const wrapper = shallow(<ZoomLevel value={100} />);
    expect(wrapper).toMatchSnapshot();
  });
});
