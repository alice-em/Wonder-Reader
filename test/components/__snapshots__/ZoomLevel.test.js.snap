// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZoomLevel should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ZoomLevel
    value={100}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": 100,
      "className": "zoomLevel",
      "style": Object {
        "cursor": "default",
        "fontFamily": "Carter One",
        "fontSize": "20px",
        "width": "45px",
      },
    },
    "ref": null,
    "rendered": 100,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": 100,
        "className": "zoomLevel",
        "style": Object {
          "cursor": "default",
          "fontFamily": "Carter One",
          "fontSize": "20px",
          "width": "45px",
        },
      },
      "ref": null,
      "rendered": 100,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
